lib/fluent/plugin/out_forward.rb

```diff
@@ -177,18 +177,14 @@ def configure(conf)
         @heartbeat_type = :transport
       end
 
-      if @dns_round_robin
-        if @heartbeat_type == :udp
-          raise Fluent::ConfigError, "forward output heartbeat type must be 'transport' or 'none' to use dns_round_robin option"
-        end
+      if @dns_round_robin && @heartbeat_type == :udp
+        raise Fluent::ConfigError, "forward output heartbeat type must be 'transport' or 'none' to use dns_round_robin option"
       end
 
       if @transport == :tls
         # socket helper adds CA cert or signed certificate to same cert store internally so unify it in this place.
         if @tls_cert_path && !@tls_cert_path.empty?
           @tls_ca_cert_path = @tls_cert_path
-        end
-        if @tls_ca_cert_path && !@tls_ca_cert_path.empty?
```

Comment:
Removing this `if` is safe?

Reply:
I found that `@tls_ca_cert_path` is defined in cofig_params. this change is wrong. reverted it.
https://github.com/fluent/fluentd/pull/2503/commits/4426b028616a4d1bc4c6245d63c003aa94117dde